build/dart/rules.gni

```diff
@@ -390,3 +390,101 @@ template("application_snapshot") {
     }
   }
 }
+
+# Generates a Dart executable using `dart compile exe`
+#
+# Arguments
+#  main_dart (required):
+#    The Dart entrypoint file.
+#
+#  package_config (required):
+#    The packages.json file for the app.
+#
+#  output (optional):
+#    The output executable name. By default, the target name. On Windows, an
+#    'exe' is automatically appended.
+#
+#  deps (optional):
+#    Additional dependencies. Dependencies on the frontend server and
+#    Flutter's platform.dill are included by default. This rule creates and
+#    uses a depfile, so listing all Dart sources is not necessary.
+template("dart_executable") {
+  assert(defined(invoker.main_dart), "Must specify 'main_dart'")
+
+  main_dart = invoker.main_dart
+
+  package_config = rebase_path(".dart_tool/package_config.json")
+  if (defined(invoker.package_config)) {
+    package_config = rebase_path(invoker.package_config)
+  }
```

Comment:
I'm going to keep `package_config` as optional, but not default it. That way this will work on code with no dependencies that doesn't need a `package_config.json` file -- e.g. single dart files etc.